cpu/stm32_common/periph/i2c_2.c

```diff
@@ -371,147 +292,115 @@ int i2c_write_bytes(i2c_t dev, uint16_t address, const void *data,
         DEBUG("[i2c] write_bytes: STOP condition was send out\n");
     }
 
-    return ret;
+    return _wait_for_bus(i2c);
 }
 
-int i2c_write_regs(i2c_t dev, uint16_t address, uint16_t reg, const void *data,
-                   size_t length, uint8_t flags)
+static int _start(I2C_TypeDef *i2c, uint8_t address_byte, uint8_t flags,
+    size_t length)
 {
-    assert(dev < I2C_NUMOF);
-
-    I2C_TypeDef *i2c = i2c_config[dev].dev;
     assert(i2c != NULL);
+    assert((i2c->SR2 & I2C_SR2_BUSY) || !(flags & I2C_NOSTART));
```

Comment:
I got a failed assertion on this line when brainlessly testing using `tests/periph_i2c` on `nucleo-l152re` with an hts221 sensor plugged. With the following commands list:
```
> i2c_acquire 0
> i2c_read_bytes 0 0x5f 1 12
```
as suggested in https://github.com/RIOT-OS/RIOT/pull/10608#issuecomment-448181648

Would it make more sense to have a return with a dedicated error value instead of crashing the application in this case ?

Reply:
Yup I don't mind that, I was trying to keep the binary size down but I can see how that would be annoying when using it (it happened to me a few times)